src/connector/src/parser/parquet_parser.rs

```diff
@@ -122,10 +123,24 @@ impl ParquetParser {
                             rw_data_type,
                         )
                     {
-                        let arrow_field =
-                            IcebergArrowConvert.to_arrow_field(rw_column_name, rw_data_type)?;
+                        let arrow_field = IcebergArrowConvert
+                            .to_arrow_field(rw_column_name, rw_data_type)
+                            .with_context(|| {
+                                format!(
+                                    "Parquet parser ERROR: to_arrow_field failed, column='{}', rw_type='{}', offset={}",
+                                    rw_column_name, rw_data_type, self.offset
+                                )
+                            })?;
                         let array_impl = IcebergArrowConvert
-                            .array_from_arrow_array(&arrow_field, parquet_column)?;
+                            .array_from_arrow_array(&arrow_field, parquet_column)
+                            .with_context(|| format!(
+                                "Parquet parser ERROR: array_from_arrow_array failed, column='{}', rw_type='{}', arrow_field='{}', parquet_type='{}', offset={}",
```

Comment:
how about making `Parquet parser ERROR` a `ConnectorError`, instead of printing the whole string.